### PR TITLE
:app — harden clothes translation normalization

### PR DESCRIPTION
**Pull request by mikelward**
### Motivation
- Prevent malformed translated prose caused by blank or whitespace-only clothing items being emitted into sentences.
- Ensure unknown/catalog-fallback items don't leak accidental surrounding whitespace into rendered output.
- Produce natural mid-sentence casing for localized garment labels (e.g. Spanish) while leaving picker labels title-cased.

### Description
- Filter out blank/whitespace entries before joining in English, German, and resource-based phrasers by mapping, trimming/translating, and `filter { it.isNotBlank() }` in `joinItems` (changes to `EnglishClothesPhraser`, `GermanClothesPhraser`, `ResourceClothesPhraser`).
- Trim unknown fallback item text in German and resource phrasers so stray whitespace is removed instead of preserved in prose.
- Lowercase localized garment labels using the active resources locale in the generic `ResourceClothesPhraser` and add a small `Resources.currentLocale()` helper to obtain the context locale.
- Add regression tests in `InsightFormatterTest` covering messy casing/whitespace Spanish catalog keys and ignoring blank clothing entries.

### Testing
- Added unit tests in `app/src/test/kotlin/app/clothescast/insight/InsightFormatterTest.kt` but could not run them end-to-end in the sandbox due to an environment Gradle/Kotlin DSL initialization failure.
- Attempts to run `./gradlew :app:testDebugUnitTest --tests "app.clothescast.insight.InsightFormatterTest"` and `./gradlew :core:domain:test` both failed early with `IllegalArgumentException: 25.0.1` from the Kotlin/Gradle tooling in this environment.
- CI should run the full Android/JVM test matrix and validate the new tests and behavior on actual build runners.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_69f115506b38832ba886a3cd329507e2)

### CODE REVIEW

File: app/src/main/kotlin/app/clothescast/insight/ClothesPhraser.kt [hunk 1]
```diff
@@ -181,7 +187,10 @@ internal class ResourceClothesPhraser(private val resources: Resources) : Clothe
...
 
     override fun joinItems(items: List<String>): String {
-        val translated = items.map(::translate)
+        val translated = items.asSequence()
+            .map(::translate)
+            .filter { it.isNotBlank() }
+            .toList()
         return when (translated.size) {
             0 -> ""
             1 -> translated[0]
```
[the remark this answers is not in the document]
Comment (mikelward) [reply app/src/main/kotlin/app/clothescast/insight/ClothesPhraser.kt:196]: Good catch — fixed in stacked PR #210 (commit d0d9331). `formatClothes`, `formatTieIn`, and `formatEveningEventTieIn` now return `String?` and emit `null` when the rendered phrase would be blank; `format()`'s `buildList` short-circuits on null, so the entire clause is dropped instead of producing `Wear .` / `Bring  tonight.`. The tie-in helpers check the input item before article picking so `prefixArticle("")` never runs.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/src/test/kotlin/app/clothescast/insight/InsightFormatterTest.kt [hunk 2]
```diff
@@ -388,4 +388,24 @@ class InsightFormatterTest {
...
+
+    @Test
+    fun `clothes clause ignores blank entries instead of speaking empty nouns`() {
+        val out = subject.format(
+            summary(
+                clothes = ClothesClause(listOf(" ", "sweater", "")),
+            ),
+        )
+        out shouldBe "Today will be mild. Wear a sweater."
+    }
```
[the remark this answers is not in the document]
Comment (mikelward) [reply app/src/test/kotlin/app/clothescast/insight/InsightFormatterTest.kt:410]: Added all three regression tests in stacked PR #210 (commit d0d9331): `clothes clause is omitted when all entries are blank or whitespace`, `calendar tie-in clause is omitted when the item is blank or whitespace`, and `evening event tie-in clause is omitted when the item is blank` (the latter passes a `rainTime` to also cover the rain-time branch). All three assert the band-only sentence, locking in the "drop the whole clause" behavior.

---
_Generated by [Claude Code](https://claude.ai/code)_
